src/documents.c

```diff
@@ -56,6 +56,7 @@ register_document (const char *uri,
                    gboolean for_save,
                    gboolean writable,
                    gboolean directory,
+                   gboolean deletable,
```

Comment:
I think this reached a point where it's better to switch to flags, e.g.:

```c
typedef enum {
  DOCUMENT_FLAG_FOR_SAVE,
  DOCUMENT_FLAG_WRITABLE,
  DOCUMENT_FLAG_DIRECTORY,
  DOCUMENT_FLAG_DELETABLE,
} DocumentFlags;
```

Reply:
I agree here, I will clean up the PR later today